examples/private/main.tf

```diff
@@ -0,0 +1,82 @@
+provider "azurerm" {
+  features {
+    resource_group {
+      prevent_deletion_if_contains_resources = false
+    }
+    key_vault {
+      purge_soft_delete_on_destroy = true
+    }
+  }
+}
+
+locals {
+  location = "uksouth"
+  tags = {
+    module  = "kubernetes-cluster"
+    example = "private"
+    usage   = "demo"
+  }
+  resource_prefix = "akc-pri-demo-uks-03"
+}
+
+resource "azurerm_resource_group" "akc" {
+  name     = "rg-${local.resource_prefix}"
+  location = local.location
+  tags     = local.tags
+}
+
+resource "azurerm_virtual_network" "akc" {
+  name                = "vnet-${local.resource_prefix}"
+  location            = local.location
+  resource_group_name = azurerm_resource_group.akc.name
+  tags                = local.tags
+
+  address_space = ["10.0.0.0/16"]
+}
+
+resource "azurerm_subnet" "akc" {
+  name                 = "snet-akc-default"
+  resource_group_name  = azurerm_virtual_network.akc.resource_group_name
+  virtual_network_name = azurerm_virtual_network.akc.name
+
+  address_prefixes  = ["10.0.0.0/20"]
+  service_endpoints = ["Microsoft.ContainerRegistry"]
+}
+
+resource "azurerm_private_dns_zone" "akc" {
+  name                = "privatelink.${local.location}.azmk8s.io"
+  resource_group_name = azurerm_resource_group.akc.name
+}
+
+resource "azurerm_private_dns_zone_virtual_network_link" "akc" {
+  name                  = azurerm_virtual_network.akc.name
+  resource_group_name   = azurerm_resource_group.akc.name
+  private_dns_zone_name = azurerm_private_dns_zone.akc.name
+  virtual_network_id    = azurerm_virtual_network.akc.id
+}
+
+data "azurerm_subscription" "current" {}
+
+module "akc" {
+  source = "../../"
+
+  name                = "akc-${local.resource_prefix}"
+  location            = local.location
+  resource_group_name = azurerm_resource_group.akc.name
+  tags                = local.tags
+
+  enable_private_cluster = true
+  // NOTE: Internally, this relies on a 'count' for assigning the role.
+  //       This is a workaround for getting the module to work properly with bring your own DNS.
+  private_dns_zone_id = format(
+    "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/privateDnsZones/privatelink.%s.azmk8s.io",
+    data.azurerm_subscription.current.subscription_id, "rg-${local.resource_prefix}", local.location
```

Comment:
Yup - default is "System" which self-manages but there's a number of sections in the code that use var.private_dns_zone_id as part of conditionals, count, ternary expressions etc.

Reply:
As mentioned below, would it be worth consolidating this approach then with the approach we've taken with Microsoft Defender and LAW? IE, having an "enabled" flag for Private DNS Zone as this would prevent the issue we are having despite it being the undesirable option.

Reply:
This should probably be discussed in more detail honestly, because we have three options. Either System, None or our own DNS Zone ID. Instead of changing this now and finding that the solution isn't adequate and missing things.

Suggest we put an item into the project board to discuss how we're handling this particular thing.